manage_breast_screening/clinics/tests/system/test_user_views_clinic_show_page.py

```diff
@@ -40,6 +40,10 @@ def test_user_views_clinic_show_page(self):
         self.then_the_appointment_is_checked_in()
         self.and_the_appointments_remain_in_the_same_order()
 
+        self.given_an_appointment_has_extra_needs()
```

Comment:
I think this might work better as a separate scenario, e.g. `test_user_views_clinic_with_special_appointments`

The test could also click through to the appointment to confirm that the banner is also shown on the appointment page.